tests/testthat/test-compat-dplyr.R

```diff
@@ -0,0 +1,263 @@
+# based on https://github.com/tidymodels/workflowsets/blob/main/tests/testthat/test-compat-dplyr.R
```

Comment:
The tests are very much yours, with some tweaks because I don't have the row uniqueness issue you have in workflowsets. I also migrated some existing tests I had of a few other dplyr verbs.